### DISCUSSION TITLE
tutorials/migrating/upgrading_to_godot_4.3

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#49] [User-contributed notes]
# tutorials/migrating/upgrading_to_godot_4.3

For most games and apps made with 4.2 it should be relatively safe to migrate to 4.3. This page intends to cover everything you need to pay attention to when migrating your project. Breaking change...

https://docs.godotengine.org/en/4.3/tutorials/migrating/upgrading_to_godot_4.3.html

**Comment by NatGazer** [comment]
If you have a Node3D parent of a top_level Node3D, moving it won't affect it's child as expected. In 4.2 the AABB box didn't account for that, now it does in 4.3 and increases as you move the parent node from it's child. On top of that, in 4.3, for some reason my Parent node teleports to it's child node position when the game begins, that made me disable top_level in editor and enable it by code in _ready() function

**Comment by NatGazer** [comment]
If you use GDExtension, keep in mind that creating bound Godot classes will crash your game from now on. Examples:

Instead of: RandomNumberGenerator rng;
Use: Ref<RandomNumberGenerator> rng; rng.instantiate();

Instead of: SphereShape3D* sphere = new SphereShape3D();
Use: SphereShape3D* sphere = memnew(SphereShape3D);

**Reply by NatGazer** [reply]
Ups, text formating removes <>. Here's the correct way:

Instead of: `SphereShape3D* sphere = new SphereShape3D();`
Use: `SphereShape3D* sphere = memnew(SphereShape3D);`

Instead of: `RandomNumberGenerator rng;`
Use: `Ref<RandomNumberGenerator> rng; rng.instantiate();`

**Reply by FireCatMagic** [reply]
`Ref<RandomNumberGenerator> rng = Ref(memnew(RandomNumberGenerator));` also works

**Comment by DiMartino49** [comment]
Where is the autoload scripts ? It seems to have disapeared from the project settings...

**Reply by DiMartino49** [reply]
Found it ! It's in Project > Project Settings > Globals

**Comment by 1Rusher** [comment]
I am creating 2D action RPG game with a top-down view, on 4.2.1. Does it make sense to upgrade to 4.3?

**Reply by Calinou** [reply]
See [Should I upgrade my project to use new engine versions?](https://docs.godotengine.org/en/latest/about/release_policy.html#should-i-upgrade-my-project-to-use-new-engine-versions).
